hive.go

```diff
@@ -62,13 +68,99 @@ func NewConnectConfiguration() *ConnectConfiguration {
 		TransportMode:        "binary",
 		HTTPPath:             "cliservice",
 		TLSConfig:            nil,
+		UseZookeeper:         false,
+		ZookeeperNamespace:   ZOOKEEPER_DEFAULT_NAMESPACE,
 	}
 }
 
 // Connect to hive server
 func Connect(host string, port int, auth string,
 	configuration *ConnectConfiguration) (conn *Connection, err error) {
 
+	if configuration.UseZookeeper {
+		// consider host as zookeeper quorum
+		zkHosts := strings.Split(host, ",")
+		for i, zkHost := range zkHosts {
+			if !strings.Contains(zkHost, ":") {
+				zkHosts[i] = fmt.Sprintf("%s:%d", zkHost, port)
+			}
+		}
+
+		zkConn, _, err := zk.Connect(zkHosts, time.Second)
+		if err != nil {
+			return nil, err
+		}
+
+		hsInfos, _, err := zkConn.Children("/" + configuration.ZookeeperNamespace)
+		if err != nil {
+			panic(err)
+		}
+		if len(hsInfos) > 0 {
+			nodes := parseHiveServer2Info(hsInfos)
+			rand.Shuffle(len(nodes), func(i, j int) {
+				nodes[i], nodes[j] = nodes[j], nodes[i]
+			})
+			for _, node := range nodes {
+				port, err := strconv.Atoi(node["port"])
+				if err != nil {
+					continue
+				}
+				conn, err := innerConnect(node["host"], port, auth, configuration)
+				if err != nil {
+					// Let's try to connect to the next one
+					continue
+				}
+				return conn, nil
+			}
+			return nil, fmt.Errorf("all Hive servers of the specified Zookeeper namespace %s are unavailable",
+				configuration.ZookeeperNamespace)
+		} else {
+			return nil, fmt.Errorf("no Hive server is registered in the specified Zookeeper namespace %s",
+				configuration.ZookeeperNamespace)
+		}
+		return nil, nil
```

Comment:
Thank you for contributing this awesome feature! I'll merge this as soon as the tests pass however they are failing now because this line is unreachable: https://travis-ci.org/github/beltran/gohive/jobs/670727851?utm_medium=notification&utm_source=github_status

Reply:
Oops, I removed it.

Reply:
I'm using this feature with https://github.com/EugeneChung/gohive/releases/tag/v1.0.6 , which it includes unreachable return. 
I should use 'go vet' for my build process, too. Thank you.